sechub-wrapper-xray/src/main/java/com/mercedesbenz/sechub/wrapper/xray/report/XrayWrapperReportSupport.java

```diff
@@ -0,0 +1,86 @@
+package com.mercedesbenz.sechub.wrapper.xray.report;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Map;
+
+import org.cyclonedx.model.Bom;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.mercedesbenz.sechub.wrapper.xray.XrayWrapperException;
+import com.mercedesbenz.sechub.wrapper.xray.XrayWrapperJSONConverter;
+import com.mercedesbenz.sechub.wrapper.xray.util.ZipFileExtractor;
+
+public class XrayWrapperReportSupport {
+
+    public record XrayReportFiles(File xrayPdsReport, File cycloneReport, File securityReport) {
+    }
+
+    public XrayReportFiles collectXrayReportsInArchive(String unzippedArchive, String pdsResultFile) throws XrayWrapperReportException {
+        File xrayPdsReport = new File(pdsResultFile);
+        ZipFileExtractor zipFileExtractor = new ZipFileExtractor();
+        File cycloneReport;
+        File securityReport;
+
+        if (!zipFileExtractor.fileExists(unzippedArchive)) {
+            // folder with reports is zipped
+            String zipArchive = unzippedArchive + ".zip";
+            if (zipFileExtractor.fileExists(zipArchive)) {
+                zipFileExtractor.unzipFile(Paths.get(zipArchive), Paths.get(unzippedArchive));
+            } else {
+                throw new XrayWrapperReportException("File with reports does not exist");
+            }
+        }
+
+        ArrayList<Path> cyclones = getFilesByName(unzippedArchive, "CycloneDX");
```

Comment:
I do not think we need the information that this is an ArrayList (and also not in next line)
```suggestion
        List<Path> cyclones = getFilesByName(unzippedArchive, "CycloneDX");
```